### PR TITLE
Add auto-registration when using http_auth

### PR DESCRIPTION
**Pull request by bpatath**
Changes proposed in this pull request:

When using `http_auth`, the app trusts the web server to perform the authentication. Thus, it should also trust the web server to only pass valid users. Currently, users must be manually created by the admin before they can log in. This PR adds an option so that, when a non-existing user authenticate using `http_auth`, the user is automatically created and logged in.

How to test the feature manually:

1. Use a working `http_auth` based setup
2. Set `http_auth_auto_register` to `true` in your config file
3. Log in to a non-existing FreshRSS user
4. The user should be created and logged in, instead of showing a permission error

Pull request checklist:

- [x] clear commit messages
- [x] code manually tested
- [ ] unit tests written (optional if too hard)
- [x] documentation updated

### CODE REVIEW

**Comment by Alkarex** [comment]
Thanks @bpatath , that sounds fair enough.
I will have a look after the release of FreshRSS 1.16.1 (in the coming days)

File: app/Models/Auth.php [hunk 1]
```diff
@@ -62,7 +62,17 @@ private static function accessControl() {
...
 			$current_user = httpAuthUser();
-			$login_ok = $current_user != '' && FreshRSS_UserDAO::exists($current_user);
+			if ($current_user == '') {
+				return false;
+			}
+			$login_ok = FreshRSS_UserDAO::exists($current_user);
+			if (!$login_ok && $conf->http_auth_auto_register) {
+				$email = null;
+				if ($conf->http_auth_auto_register_email_field !== '') {
+					$email = $_SERVER[$conf->http_auth_auto_register_email_field];
```
Comment (Alkarex) [inline app/Models/Auth.php]:
```suggestion
				if ($conf->http_auth_auto_register_email_field != '' &&
					isset($_SERVER[$conf->http_auth_auto_register_email_field])) {
					$email = $_SERVER[$conf->http_auth_auto_register_email_field];
```

We probably need to check that this key exists

**Comment by Alkarex** [comment]
@plopoyop You are using HTTP Auth in YunoHost; do you have any feedback on this user auto-create feature? (probably not useful for YunoHost but you might have an opinion anyway)

**Comment by Alkarex** [comment]
@bpatath Could you please add a little note in our documentation? https://github.com/FreshRSS/FreshRSS/blob/master/docs/en/admins/09_AccessControl.md

**Comment by plopoyop** [comment]
> probably not useful for YunoHost

We are calling freshrss user creation script every time a  user is created in yunohost.
Auto registration would be great.

**Comment by Alkarex** [comment]
Ok @plopoyop so as this pull request is, would it be directly helpful, or do you need anything else?

**Comment by marienfressinaud** [comment]
Do we need an option for that? It seems fine to me to have the feature always enabled

**Comment by Alkarex** [comment]
> Do we need an option for that? It seems fine to me to have the feature always enabled

Indeed, as we can whitelist / blacklist users at Web server level (e.g. `.htaccess`), but I was afraid I might had overlooked an undesirable security implication

**Comment by Frenzie** [comment]
That seems reasonable given the context, but I'm not quite sure.

**Comment by marienfressinaud** [comment]
Maybe if a server has several users but the sysadmin wants to restrict who can access FRSS? Indeed we can keep the option for this reason (but is a sysadmin really manage its users this way? :p) It looks more admin-friendly to switch it on by default though

Edit: sorry, misclick

**Comment by Alkarex** [comment]
> Maybe if a server has several users but the sysadmin wants to restrict who can access FRSS?

For that an `allow` or `deny` style of rule at Web server level can also do the job

**Comment by Alkarex** [comment]
@bpatath I have made a few changes. Could you please double-check that it still works for you?

File: app/Models/Auth.php [hunk 2]
```diff
@@ -47,8 +47,8 @@ public static function init() {
 	 * @return boolean true if user can be connected, false else.
 	 */
 	private static function accessControl() {
-		$conf = Minz_Configuration::get('system');
-		$auth_type = $conf->auth_type;
+		FreshRSS_Context::$system_conf = Minz_Configuration::get('system');
```
Comment (Alkarex) [inline app/Models/Auth.php:50]: I will make another PR with a better refactoring of this part, but which requires a few global changes.

**Comment by Alkarex** [comment]
Let's continue testing in /master

File: app/Models/Auth.php (hunk 2 above)
Comment (Alkarex) [reply app/Models/Auth.php:50]: https://github.com/FreshRSS/FreshRSS/pull/3070

**Pull request merged**